server/src/main/java/org/elasticsearch/action/ActionModule.java

```diff
@@ -460,10 +460,10 @@ public ActionModule(boolean transportClient, Settings settings, IndexNameExpress
                     throw new IllegalArgumentException("Cannot have more than one plugin implementing a REST wrapper");
                 }
                 restWrapper = newRestWrapper;
-                if (restWrapper.getClass().getCanonicalName() == null ||
-                    restWrapper.getClass().getCanonicalName().startsWith("org.elasticsearch") == false) {
+                if (plugin.getClass().getCanonicalName() == null ||
+                    plugin.getClass().getCanonicalName().startsWith("org.elasticsearch") == false) {
```

Comment:
On the call earlier, we proposed actually checking whether the plugin is the security plugin, rather than just looking at the package prefix.
Is there a reason not to do that?

```suggestion
                if ("org.elasticsearch.xpack.security.Security".equals(plugin.getClass().getCanonicalName()) == false) {
```